domain/schema/controller.go

```diff
@@ -347,3 +350,95 @@ BEGIN
     VALUES (4, 2, OLD.controller_id, DATETIME('now'));
 END;`[1:]
 }
+
+func modelMigrationSchema() string {
+	return `
+CREATE TABLE model_migration (
+    uuid                    TEXT PRIMARY KEY,
+    attempt                 INT,
+    target_controller_uuid  TEXT NOT NULL,
+    target_entity           TEXT,
+    target_password         TEXT,
+    target_macaroons        TEXT,
+    active                  BOOLEAN,
+    start_time              TIMESTAMP,
+    success_time            TIMESTAMP,
+    end_time                TIMESTAMP,
+    phase                   TEXT,
+    phase_changed_time      TIMESTAMP,
+    status_message          TEXT,
+    CONSTRAINT              fk_model_migration_target_controller
+        FOREIGN KEY         (target_controller_uuid)
+        REFERENCES          external_controller(uuid)
+);
+
+CREATE TABLE model_migration_status (
+    uuid                TEXT PRIMARY KEY,
+    start_time          TIMESTAMP,
+    success_time        TIMESTAMP,
+    end_time            TIMESTAMP,
+    phase               TEXT,
+    phase_changed_time  TIMESTAMP,
+    status              TEXT
+);
+
+CREATE TRIGGER trg_log_model_migration_status_insert
+AFTER INSERT ON model_migration_status FOR EACH ROW
+BEGIN
+    INSERT INTO change_log (edit_type_id, namespace_id, changed_uuid, created_at) 
+    VALUES (1, 4, NEW.key, DATETIME('now'));
+END;
+CREATE TRIGGER trg_log_model_migration_status_update
+AFTER UPDATE ON model_migration_status FOR EACH ROW
+BEGIN
+    INSERT INTO change_log (edit_type_id, namespace_id, changed_uuid, created_at) 
+    VALUES (2, 4, OLD.key, DATETIME('now'));
+END;
+CREATE TRIGGER trg_log_model_migration_status_delete
+AFTER DELETE ON model_migration_status FOR EACH ROW
+BEGIN
+    INSERT INTO change_log (edit_type_id, namespace_id, changed_uuid, created_at) 
+    VALUES (4, 4, OLD.key, DATETIME('now'));
+END;
+
+CREATE TABLE model_migration_user (
```

Comment:
I think this is the last thing. You will need a migration_uuid column here with a foreign key constraint to model_migration.

Reply:
Done